pkg/pool-weighted/contracts/managed/ManagedPool.sol

```diff
@@ -1081,7 +1089,7 @@ contract ManagedPool is BaseWeightedPool, ProtocolFeeCache, ReentrancyGuard, ICo
         uint256 endSwapFeePercentage
     ) internal virtual {
         if (startSwapFeePercentage != getSwapFeePercentage()) {
-            super._setSwapFeePercentage(startSwapFeePercentage);
```

Comment:
Whoah, this changes behavior, doesn't it? With this change, we'd no longer allow a gradual change if the start value is not the current one _and_ there's an ongoing change.

It's also interesting that with this change we potentially write the end time twice, since both here and in the call we call `_setSwapFeeData()`. At any rate, the whole structure is confusing. It'd be great to eventually get rid of `setSwapFeePercentage()` entirely.

Reply:
> Whoah, this changes behavior, doesn't it? With this change, we'd no longer allow a gradual change if the start value is not the current one and there's an ongoing change.

Ah this is true, I'll address this later.

Reply:
I've hoisted up `setSwapFeePercentage` to ManagedPool to hold the behaviour blocking changing the swap fee when there's a gradual update ongoing.

Agreed on the structure being confusing. Once `ManagedPoolStorageLib` is in then we can refactor this to only do a single write.